File: medium/1769_minimum_number_of_operations_to_move_all_balls_to_each_box/solution.cpp
#include <numeric>
#include <string>
#include <vector>

class Solution {
public:
  std::vector<int> minOperations(std::string boxes) {
    const size_t n = boxes.size();

    std::vector<int> onesIndices;

    for (size_t i{}; i < n; ++i) {
      if (boxes[i] == '1')
        onesIndices.push_back(static_cast<int>(i));
    }

    std::vector<int> operations(n, 0);

    for (size_t i{}; i < n; ++i) {
      operations[i] = std::accumulate(
          onesIndices.begin(), onesIndices.end(), 0,
          [&](int acc, int j) { return acc + abs(static_cast<int>(i) - j); });
    }

    return operations;
  }
};

File: medium/1769_minimum_number_of_operations_to_move_all_balls_to_each_box/solution.py
class Solution:
    def minOperations(self, boxes: str) -> list[int]:
        n = len(boxes)

        acc = 0
        ones = 0

        operations = [0] * n

        for i in range(n):
            acc += ones
            operations[i] = acc

            if boxes[i] == "1":
                ones += 1

        acc = 0
        ones = 0

        for i in range(n - 1, -1, -1):
            acc += ones
            operations[i] += acc

            if boxes[i] == "1":
                ones += 1

        return operations
